src/main/java/com/minecolonies/coremod/colony/buildings/AbstractCitizenAssignable.java

```diff
@@ -0,0 +1,255 @@
+package com.minecolonies.coremod.colony.buildings;
+
+import com.minecolonies.api.colony.requestsystem.requester.IRequester;
+import com.minecolonies.api.colony.requestsystem.resolver.IRequestResolverProvider;
+import com.minecolonies.coremod.colony.CitizenData;
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.entity.EntityCitizen;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.math.BlockPos;
+import net.minecraftforge.fml.common.gameevent.TickEvent;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Abstract class handling requests from the building side.
+ */
+public abstract class AbstractCitizenAssignable extends AbstractSchematicProvider implements IRequestResolverProvider, IRequester
+{
+    /**
+     * The colony the building belongs to.
+     */
+    @NotNull
+    protected final Colony colony;
+
+    /**
+     * List of workerbuildings assosiated to the building.
+     */
+    private final List<CitizenData> assignedCitizen = new ArrayList();
+
+    /**
+     * Constructor for the abstract class which receives the position and colony.
+     * @param pos
+     * @param colony
+     */
+    public AbstractCitizenAssignable(final BlockPos pos, final Colony colony)
+    {
+        super(pos);
+        this.colony = colony;
+    }
+
+    @Override
+    public void readFromNBT(@NotNull final NBTTagCompound compound)
+    {
+        super.readFromNBT(compound);
+        assignedCitizen.clear();
+    }
+
+    @Override
+    public void writeToNBT(@NotNull final NBTTagCompound compound)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this method to simply inherit it. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1185)